satpy/readers/hdfeos_base.py

```diff
@@ -401,7 +360,10 @@ def get_interpolated_dataset(self, name1, name2, resolution, offset=0):
             result2 = self.cache[(name2, resolution)]
         except KeyError:
             result1 = self._load_ds_by_name(name1)
-            result2 = self._load_ds_by_name(name2) - offset
+            result2 = self._load_ds_by_name(name2)
+            if offset != 0:
+                # avoid extra dask tasks if not necessary
+                result2 = result2 - offset
```

Comment:
you are right when you mention that the offset in confusing. I think we should refactor the interpolation a bit:
- how about having two methods on top of this one, one called eg `get_interpolated_lonlats` and the other `get_interpolated_angles`, reading the data and the latter taking care of the offset subtraction and addition?
- then this method doesn’t need `offset` anymore
- as for the caching, can we use lru_cache somehow?
I know this is maybe not in the scope of the original PR, but I would help with cleanliness :)

Reply:
To be clear, the main confusion was when looking at the dask graph as an SVG and following the path of reflectance data, lon/lat arrays, and angles, and seeing a "random" subtraction task.

For those two interpolated methods, it feels like it would require some duplicate code but that may be easier than something else.

For caching with a cache decorator, my fear is always doing it right on a class method to not cache the class instance itself. Let me spend 5 minutes checking the current state of that stuff.

Reply:
What you provided in your last commit covers what I had in mind actually